kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -475,6 +467,21 @@ TableInfo constructTableInfo(TableId table, com.google.cloud.bigquery.Schema big
     return tableInfoBuilder.build();
   }
 
+  private TimePartitioning setTimePartitioningForDefinition(TableId table, StandardTableDefinition.Builder builder) {
+    TimePartitioning timePartitioning = TimePartitioning.of(Type.DAY); // DEFAULT is DAY
+    Table bigQueryTable = bigQuery.getTable(table);
```

Comment:
This method can be called in both the case of creating a table and updating one, correct? At least in the first case, does it make sense to even try to perform the lookup here? And if we're updating a table, do we have to explicitly set a time partitioning type? Have you tried leaving the partitioning type unset entirely and seeing if the existing partitioning type remains in use?

Reply:
I think this still needs to be addressed? Specifically:

- If a table is being created instead of updated, we shouldn't look up the partitioning type for the table since it shouldn't exist in the first place (of course, there's always the possibility of a race condition with other tasks and/or manual actions on the part of the user, but a lookup here won't prevent those in the case that the other tasks or the user act after the lookup has been performed).
- If a table is being updated instead of created, have you tried leaving the partitioning type unset entirely and seeing if the existing partitioning type remains in use?

Reply:
This method has been renamed to getTimePartitioningForDefinition(tableId), its current usage is only from within the constructTableInfo method. In the original code the time partitioning was being hard-coded to DAY. As requested for the intermediateTables I have reverted it to DAY, but for the non intermediate tables I felt that we should do the look up and if none is found it will be set to DAY as the default as was in the original codebase.

The returned time partitioning is utilised to set the timestampPartitionFieldName field and in the case of clusteringFieldName field if clustering is present. If I were to remove this method, how do you propose we go about setting the values for above mentioned fields?

Reply:
> for the non intermediate tables I felt that we should do the look up

What purpose does the lookup serve, though? If we're creating a table from scratch, the lookup will fail and we won't get any relevant information from it. If we're updating an existing table, it won't be possible to alter the partitioning type anyway, so we could try leaving it unset in our schema update request altogether (which should be fine if BigQuery's API allows this).

With the case of timestamp field names, I think the same logic applies: if we're creating a table, we stick to the default of `DAY` time partitioning and provide the timestamp column name; if we're updating a table, we leave out partitioning information altogether and see it if remains unchanged since partitioning types for tables can't be updated after they're created anyways.

With the case of table clustering, it's possible that the same logic applies as well. The only difference I can see is that the clustering style for a table can be updated after it's created (see https://cloud.google.com/bigquery/docs/creating-clustered-tables#modifying-cluster-spec), but if we simply refrain from specifying a clustering style during a schema update, it's still possible that the current clustering style for the table will be preserved.

Reply:
@mubeen I've gotten around to doing some testing here and there's some good news and some bad news.

Good news: sending a table schema update request with no time partitioning specified succeeds, and doesn't remove the existing partitioning type for the table (which makes since since BigQuery doesn't support changing the partitioning type for existing tables anyways).

However, sending a table schema update request with a different time partitioning type than the one the table uses does not succeed and results in a 400 response from BigQuery.

So, I believe we need to alter the `SchemaManager::constructTableInfo` method to not include a time partitioning type at all when  a schema is being updated, and instead only include one when a new table is being created.